PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStoragePanel.kt

```diff
@@ -143,6 +153,52 @@ open class SparkSubmissionJobUploadStoragePanel: JPanel(), Disposable {
         var deployStorageTypesModel: ComboBoxModel<SparkSubmitStorageType> by ComboBoxModelDelegated(storageTypeComboBox)
 
         val storageCheckSubject: PublishSubject<StorageCheckEvent> = disposableSubjectOf { PublishSubject.create() }
+
+        fun prepareVFSRoot(uploadRootPath: String?, storageAccount: IHDIStorageAccount?): AzureStorageVirtualFile? {
```

Comment:
If `uploadRootPath` and `storageAccount` are not allowed to be `null`, why just declare their type to `String` and `IHDIStorageAccount`?

Reply:
it can be null and we give warning.

---
In reply to: [297081281](https://github.com/microsoft/azure-tools-for-java/pull/3177#discussion_r297081281) [](ancestors = 297081281)